app/packs/javascripts/dynamic_elements/topic.js

```diff
@@ -171,6 +171,44 @@ export default class Topic extends ShikiEditable {
         this.$commentsHider.show();
       }
     });
+
+    this._bindReturnClick();
+  }
+
+  _showReturnButton() {
+    if ($('.b-comments').height() > $(window).height()) {
+      $('#return-to-reply').show();
+    }
```

Comment:
проверка не всегда правильно срабатывает. давай проверять на то, видно ли место клика на странице
```suggestion
  async _showReturnButton(repliedNode) {
    await delay(300); // нужно подождать, пока страница проскролится до редактора
    
    if (!$(clickedNode).is(':appeared')) {
      $('#return-to-reply').show();
    }
  }
```